### PR TITLE
Refactor dataservice and enrichmentservice part 1

### PR DESCRIPTION
**Pull request by defong**
### Context
Dataservice and Enrichmentservice does not fit into our overall design for FatControllers

Part 1 of xxx

### Changes proposed in this pull request

Removed 2 of 5 methods from DataService

### Guidance to review
Ignore `OrganisationController` inject of `EnrichmentService`

### CODE REVIEW

File: tests/ManageCourses.Tests/UnitTesting/DataServiceTests.cs [hunk 1]
```diff
@@ -29,39 +29,39 @@ public void SetUp()
...
-            var publishedOrg = _dataService.GetInstitutionSummaryForUser(email, instCode);
-            publishedOrg.Should().NotBeNull();
-            publishedOrg.EnrichmentWorkflowStatus.Should().Be(ucasInstitutionEnrichmentGetModel.Status, $"there's a {ucasInstitutionEnrichmentGetModel.Status} enrichment present");
-        }
+        //     // test published
+        //     ucasInstitutionEnrichmentGetModel.Status = EnumStatus.Published;
+        //     var publishedOrg = _dataService.GetOrganisationForUser(email, instCode);
+        //     publishedOrg.Should().NotBeNull();
+        //     publishedOrg.EnrichmentWorkflowStatus.Should().Be(ucasInstitutionEnrichmentGetModel.Status, $"there's a {ucasInstitutionEnrichmentGetModel.Status} enrichment present");
+        // }
```
Comment (timabell) [inline tests/ManageCourses.Tests/UnitTesting/DataServiceTests.cs]: kill commented out test or uncomment it and fix it

File: tests/ManageCourses.Tests/DbIntegration/DataServiceTests.cs [hunk 2]
```diff
@@ -35,118 +35,118 @@ protected override void Setup()
...
-            {
-                var result = DataService.GetCoursesForUser(TestUserEmail1, org.InstCode);//get the course for each org
-
-                foreach (var course in result)
-                {
-                    course.CourseSites.All(s => s.Publish == "N").Should().BeTrue();
-                }
-            }
-        }
+        // [Test]
```
Comment (timabell) [inline tests/ManageCourses.Tests/DbIntegration/DataServiceTests.cs]: kill commented out tests or uncomment it and fix them

**Comment by defong** [comment]
I know, but as a stop gap its going to be in there till we got rid of it both

File: src/ManageCourses.Api/Controllers/OrganisationsController.cs [hunk 3]
```diff
@@ -44,22 +48,61 @@ public ActionResult Get(string instCode)
...
+                    InstCode = institutionSummary.Institution.InstCode,
+                    TotalCourses = institutionSummary.Institution.Courses.Select(c => c.CourseCode).Distinct().Count()
+                }).OrderBy(x => x.InstName).ToList();
+
+            return institutionSummaries;
+        }
+
+        private InstitutionSummary Mapping(OrganisationInstitution organisationInstitution, string email, string instCode) {
+
+        if (organisationInstitution != null)
```
Comment (timabell) [inline src/ManageCourses.Api/Controllers/OrganisationsController.cs]: indentation is off, reformat

Comment (timabell) [reply src/ManageCourses.Api/Controllers/OrganisationsController.cs]: invert `if` to reduce nesting

File: tests/ManageCourses.Tests/DbIntegration/Controllers/OrganisationsControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GovUk.Education.ManageCourses.Tests.DbIntegration
+{
+    [TestFixture]
+    [Category("Integration")]
+    [Category("Integration_DB")]
+    // [Explicit]
```
Comment (timabell) [inline tests/ManageCourses.Tests/DbIntegration/Controllers/OrganisationsControllerTests.cs]: kill if not needed

File: tests/ManageCourses.Tests/DbIntegration/Controllers/OrganisationsControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+            organisationsController.SetControllerContext(UnauthorisedUserEmail);
+
+            var result = organisationsController.GetAll();
+
+            result.Should().BeOfType<UnauthorizedResult>();
+            result.Should().NotBeOfType<OkResult>();
+        }
+
+        [Test]
+        public void Get_Ok()
```
Comment (timabell) [inline tests/ManageCourses.Tests/DbIntegration/Controllers/OrganisationsControllerTests.cs]: `GetAll_Returns_InstNames` would be clearer and more accurate

File: src/ManageCourses.Api/Services/Data/IDataService.cs [hunk 6]
```diff
@@ -6,11 +6,11 @@ namespace GovUk.Education.ManageCourses.Api.Data
 {
     public interface IDataService
     {
-        IEnumerable<InstitutionSummary> GetInstitutionSummariesForUser(string email);
-        InstitutionSummary GetInstitutionSummaryForUser(string email, string instCode);
+
+        // IEnumerable<InstitutionSummary> GetInstitutionSummariesForUser(string email);
+        // InstitutionSummary GetInstitutionSummaryForUser(string email, string instCode);
```
Comment (timabell) [inline src/ManageCourses.Api/Services/Data/IDataService.cs]: kill commented out code

**Comment by timabell** [comment]
Well shit I wish github would tell me it's already gone, or tell everyone that I'm looking at it.

**Comment by defong** [comment]
[Refactor DataService and EnrichmentService](https://trello.com/c/XjIJ2ktw/534-refactor-dataservice-and-enrichmentservice)

**Pull request merged**